src/transformState.js

```diff
@@ -5,7 +5,29 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      Object.assign(state, obj.extraData);
+    }
+
+    if (obj.type === 'removeProperties') {
+      for (const key in state) {
+        for (const ch of obj.keysToRemove) {
+          if (key === ch) {
+            delete state[key];
+          }
+        }
```

Comment:
you don't need two loops here
make one loop for `obj.keysToRemove` and delete every key from `state` witch presented in `obj.keysToRemove`


also, you don't need `if`
If key doesn't exist in object and you are trying to remove this key, JS will skip it without any error
<img width="225" alt="Screenshot 2022-04-26 at 20 48 35" src="https://user-images.githubusercontent.com/37452763/165361353-ab345f77-fa02-40f4-8a3f-1d2a34e9ae3d.png">